packages/extension-image/src/image.ts

```diff
@@ -96,4 +112,82 @@ export const Image = Node.create<ImageOptions>({
       }),
     ]
   },
+
+  addProseMirrorPlugins() {
+    return [
+      new Plugin({
+        key: new PluginKey('imageHandler'),
+        props: {
+          handlePaste: (view, event) => {
+            if (!this.options.allowPasteFromClipboard) {
+              return false
+            }
+            const items = Array.from(event.clipboardData?.items || [])
+
+            event.preventDefault()
+
+            items.forEach(item => {
+              if (item.type.indexOf('image') === 0) {
+                const { schema } = view.state
+
+                const image = item.getAsFile()
+
+                this.options.imageUpload?.(image).then(src => {
+                  const node = schema.nodes.image.create({
+                    src,
```

Comment:
adding an option for custom attributes like a file id for downstream extraction/relationships would be money